src/components/QueryEditor/QueryEditor.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Spinner } from '@grafana/ui';
+import { useGroups } from './useGroups';
+import { QueryEditorForm, XrayQueryEditorFormProps } from './QueryEditorForm';
+
+/**
+ * Simple wrapper that is only responsible to load groups and delay actual render of the QueryEditorForm. Main reason
+ * for that is that there is queryInit code that requires groups to be already loaded and is separate hook wo it
+ * cannot be inside a condition. There are other ways to put it into single component but this seems cleaner than
+ * alternatives.
+ * @param props
+ * @constructor
```

Comment:
```suggestion
```
Is this needed here?

Reply:
No just Goland autocompleting comments a bit too much.